platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py

```diff
@@ -40,103 +58,106 @@
+        else:
+            eeprom_params = platfrom_eeprom_to_params(get_platform_eeprom())
+            stdout_stream = sys.stdout
+            sys.stdout = open(os.devnull, 'w')
+            self._eeprom_bin = self.set_eeprom(self._eeprom_bin, [eeprom_params])
+            sys.stdout.close()
+            sys.stdout = stdout_stream
         try:
-            eeprom_data = eeprom_tlvinfo.TlvInfoDecoder.set_eeprom(self, "", [eeprom_params])
-        finally:
-            decode_output = sys.stdout.getvalue()
-            sys.stdout = orig_stdout
-
-        eeprom_base.EepromDecoder.write_eeprom(self, eeprom_data)
-        self.__eeprom_tlv_dict = self.__parse_output(decode_output)
-
-    def __parse_output(self, decode_output):
-        EEPROM_DECODE_HEADLINES = 6
-        lines = decode_output.replace('\0', '').split('\n')
-        lines = lines[EEPROM_DECODE_HEADLINES:]
-        res = dict()
+            self.write_eeprom(self._eeprom_bin)
+            self.report_status("ok")
+        except IOError as e:
+            logging.error("Failed to write eeprom : %s" % (str(e)))
 
-        for line in lines:
-            try:
-                # match whitespace-separated tag hex, length and value (value is mathced with its whitespaces)
-                match = re.search('(0x[0-9a-fA-F]{2})([\s]+[\S]+[\s]+)([\S]+[\s]*[\S]*)', line)
-                if match is not None:
-                    code = match.group(1)
-                    value = match.group(3).rstrip('\0')
-                    res[code] = value
-            except Exception:
-                pass
-        return res
+        visitor = SavingTlvVisitor()
+        self.visit_eeprom(self._eeprom_bin, visitor)
+        self._system_eeprom_info = visitor.get_contents()
 
-    def __tlv_get(self, code):
-        return self.__eeprom_tlv_dict.get("0x{:X}".format(code), 'N/A')
+    def get_contents(self):
+        return self._eeprom_bin, self._system_eeprom_info
 
-    def system_eeprom_info(self):
-        return self.__eeprom_tlv_dict
-
-    def serial_number_str(self):
-        return self.__tlv_get(self._TLV_CODE_SERIAL_NUMBER)
-
-    def serial_str(self):
-        return self.serial_number_str()
-
-    def base_mac_addr(self):
-        return self.__tlv_get(self._TLV_CODE_MAC_BASE)
-
-    def part_number_str(self):
-        return self.__tlv_get(self._TLV_CODE_PART_NUMBER)
-
-    def modelstr(self):
-        return self.__tlv_get(self._TLV_CODE_PRODUCT_NAME)
-
-    def revision_str(self):
-        return self.__tlv_get(self._TLV_CODE_LABEL_REVISION)
+    def report_status(self, status):
+        status_file = None
+        try:
+            status_file = open(_EEPROM_STATUS, "w")
+            status_file.write(status)
+        except IOError as e:
+            logging.error("Failed to report state : %s" % (str(e)))
+        finally:
+            if status_file is not None:
+                status_file.close()
 
+class SavingTlvVisitor(eeprom_tlvinfo.EepromDefaultVisitor):
```

Comment:
Please rename to something like `EepromContentVisitor` or `EepromDataVisitor` or...

Reply:
Done